### PR TITLE
add bbox check to check dataset script

### PR DESCRIPTION
**Pull request by pj-ms**

### CODE REVIEW

**Review by hanxiaotian — APPROVED** [review]

File: vision_datasets/commands/check_dataset.py [hunk 1]
```diff
@@ -103,16 +121,17 @@ def main():
         logger.info(f'{prefix} Check dataset with usage: {usage}.')
 
         # if args.local_dir is none, then this check will directly try to access data from azure blob. Images must be present in uncompressed folder on azure blob.
-        dataset = dataset_hub.create_manifest_dataset(container_sas=args.blob_container, local_dir=args.local_dir, name=dataset_info.name, version=args.version, usage=usage)
+        dataset = dataset_hub.create_manifest_dataset(container_sas=args.blob_container, local_dir=args.local_dir, name=dataset_info.name, version=args.version, usage=usage, coordinates='absolute')
         if dataset:
+            if args.data_type in [DatasetTypes.IC_MULTICLASS, DatasetTypes.IC_MULTILABEL, DatasetTypes.OD]:
+                classification_detection_check(dataset)
```
Comment (cy-bai) [inline vision_datasets/commands/check_dataset.py]: pass "err_msg_file" here?

File: vision_datasets/commands/check_dataset.py [hunk 2]
```diff
@@ -49,14 +49,32 @@ def check_images(dataset: ManifestDataset, err_msg_file: pathlib.Path):
...
+    for sample_idx, sample in enumerate(dataset.dataset_manifest.images):
         labels = sample.labels
         c_ids = set([label[0] if dataset.dataset_info.type == DatasetTypes.OD else label for label in labels])
         for c_id in c_ids:
             n_imgs_by_class[c_id] += 1
 
+        if dataset.dataset_info.type == DatasetTypes.OD:
+            w, h = sample.width, sample.height
+            if not w or not h or w < 0 or h < 0:
+                err_msg_file.write_text(f'Image {sample_idx} has invalid width or height: {w}, {h}', encoding='utf-8')
```
Comment (cy-bai) [inline vision_datasets/commands/check_dataset.py]: this way only the last log will be written? need append mode

File: vision_datasets/commands/check_dataset.py [hunk 3]
```diff
@@ -103,16 +125,18 @@ def main():
...
+                errors.extend(classification_detection_check(dataset))
+
             if args.quick_check:
                 quick_check_images(dataset)
             else:
-                check_images(dataset, err_msg_file)
-
-            if args.data_type in [DatasetTypes.IC_MULTICLASS, DatasetTypes.IC_MULTILABEL, DatasetTypes.OD]:
-                classification_detection_check(dataset)
+                errors.extend(check_images(dataset, err_msg_file))
```
Comment (cy-bai) [inline vision_datasets/commands/check_dataset.py]: `check_images` doesn't take `err_msg_file`, please test the code.

Comment (pj-ms) [reply vision_datasets/commands/check_dataset.py]: Occupied with 2 Cx impacted fixes right now. Please work with Pei based on this PR and submit necessary fixes if needed, as it is purposed as a quick base indicator for fixing the CPG dataset.

If no extra bandwidth, I can better test this and finish the PR early next week.

Comment (cy-bai) [reply vision_datasets/commands/check_dataset.py]: no worries, we will work on it

**Review by cy-bai — APPROVED** [review]
LGTM thanks！

**Pull request merged**